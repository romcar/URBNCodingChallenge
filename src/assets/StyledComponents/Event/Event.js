import styled from 'styled-components';
import colors from '../../theme';

export default styled.div`
  height: 32vh;
  max-width: 28%;
  box-shadow: 0.25em 0.25em 0.25em ${colors.shadow};
  display: inline-block;
  margin: 0.5em 0;
  padding: 0.5em 20px;
  border: 1px solid ${colors.shadow};

  display: inline-grid;
  grid-template-rows: repeat(12, 1fr);
  grid-template-columns: repeat(9, 1fr);

  @media screen and (max-width: 825px) {
    max-width: 40%;
  }

  `;
