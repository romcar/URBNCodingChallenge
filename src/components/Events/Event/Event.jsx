import React, { Component } from 'react';
import StyledEvent from '../../../assets/StyledComponents/Event/Event';
import utils from '../../../assets/utils';
import moment from 'moment';
// styled components
import StyledImage from '../../../assets/StyledComponents/Image/Image';
import StyledEventInfo from '../../../assets/StyledComponents/EventInfo/EventInfo';
import StyledEventTitle from '../../../assets/StyledComponents/EventInfo/EventTitle';
import StyledEventLocation from '../../../assets/StyledComponents/EventInfo/EventLocation';

class Event extends Component {
  constructor(props) {
    super(props);
  }

  toggleEventModal() {
    console.log('toggling');
  }


  render() {
    const { event } = this.props;
    const convert = utils.convertSpecialCharsToString;

    return (
      <StyledEvent className="event">
        <StyledEventInfo className="event-item" onClick={this.toggleEventModal}>
          <div className="event-image">
            {event.image ? <StyledImage src={event.image.medium.url} /> : null}
          </div>
          <div className="event-info">
            <StyledEventTitle className="event-title">{event.title}</StyledEventTitle><StyledEventLocation className="event-loc">{`${event.city_name}, ${event.region_abbr}. ${moment(event.start_time).format('llll')}`}</StyledEventLocation>
            <br />
            <StyledEventLocation className="event-venue">{event.venue_name}</StyledEventLocation>
            <br />
            <span className="event-desc">{event.description ? convert(event.description.substring(0, 200)) : 'There is no description for this event, sorry'}</span>
          </div>
        </StyledEventInfo>
      </StyledEvent>
    );
  }
}

export default Event;