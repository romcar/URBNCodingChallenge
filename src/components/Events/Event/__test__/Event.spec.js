import React from 'react';
import testData from '../../../../../fakeSeed';
import { configure, shallow } from 'enzyme';
import Event from '../Event';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import moment from 'moment';

configure({ adapter: new Adapter() });

let testEvent = null;
describe('Event Component', () => {
  beforeEach(() => {
    testEvent = testData[0];
  });

  it('Should render component', () => {
    // console.log(testData)
    const wrapper = shallow(<Event event={testEvent} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.event').length).toEqual(1);
  });

  it('Should have an image', () => {
    const wrapper = shallow(<Event event={testEvent} />)
    expect(wrapper.find('.event-image').length).toEqual(1);
  });

  it('Should have some information', () => {
    const wrapper = shallow(<Event event={testEvent} />);
    expect(wrapper.find('.event-info').length).toEqual(1);
    // includes the break tags
    expect(wrapper.find('.event-info').children().length).toEqual(6);
  });

  it('Should have an event title', () => {
    const wrapper = shallow(<Event event={testEvent} />);
    expect(wrapper.find('.event-title').render().text()).toEqual(testEvent.title);
  });

  it('Should have the location and time of the event', () => {
    const wrapper = shallow(<Event event={testEvent} />);
    // time is converted to something readable using moment js
    // I trust they know what they are doing.
    const expected = `${testEvent.city_name}, ${testEvent.region_abbr}. ${moment(testEvent.start_time).format('llll')}`
    expect(wrapper.find('.event-loc').render().text()).toEqual(expected);
  });

  it('Should have the event venue displayed', () => {
    const wrapper = shallow(<Event event={testEvent} />);

    expect(wrapper.find('.event-venue').length).toEqual(1);
    expect(wrapper.find('.event-venue').render().text()).toEqual(testEvent.venue_name);
  });

  it('Should have an event description, if one is available', () => {
    const wrapper = shallow(<Event event={testEvent} />);
    expect(wrapper.find('.event-desc').length).toEqual(1);

    if (testEvent.description) {
      let expected = testEvent.description.substring(0, 200);
      expect(wrapper.find('.event-desc').render().text()).toEqual(expected);
    } else {
      expect(wrapper.find('.event-desc').render().text()).toEqual("There is no description for this event, sorry");
    }
  });

  it('Should have an event description that does not exceed 200 chars', () => {
    const wrapper = shallow(<Event event={testEvent} />);
    expect(wrapper.find('.event-desc').render().text().length).toBeLessThanOrEqual(200);
  });

});
