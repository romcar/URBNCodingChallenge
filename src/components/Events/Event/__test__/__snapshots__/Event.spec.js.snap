// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component Should render component 1`] = `
<styled.div
  className="event"
>
  <styled.div
    className="event-item"
    onClick={[Function]}
  >
    <div
      className="event-image"
    >
      <styled.img
        src="http://d1marr3m5x4iac.cloudfront.net/images/medium/I0-001/003/948/039-3.jpeg_/sugarland-39.jpeg"
      />
    </div>
    <div
      className="event-info"
    >
      <styled.h4
        className="event-title"
      >
        Sugarland
      </styled.h4>
      <styled.span
        className="event-loc"
      >
        Philadelphia, PA. Sun, Sep 9, 2018 7:00 PM
      </styled.span>
      <br />
      <styled.span
        className="event-venue"
      >
        Mann Center for the Performing Arts
      </styled.span>
      <br />
      <span
        className="event-desc"
      >
        There once was a description here but no more. Actually there was no description and a story had to be told. He picked up the watter bottle, took a sip, and put it down. Then immediately took a drag o
      </span>
    </div>
  </styled.div>
</styled.div>
`;
